### DISCUSSION TITLE
Consistent keyboard shortcuts

### DISCUSSION THREAD

**Discussion by grahamperrin** [ravynsoft/ravynos#137] [General]
As a user of KDE Plasma and other desktop environments on FreeBSD and Linux, and Windows 10: 

* I **love** that things such as <kbd>Alt</kbd>-<kbd>F4</kbd> work as expected

– there's a sense of consistency. 

From <https://airyx.org/>: 

> Consistent keyboard shortcuts and menus

@mszoek please, will Airyx attempt to make Apple keyboard behaviours a default? (I hope not.)

<https://airyx.org/faq.html> make most things clear, but there's no mention of keyboards. 

TIA

**Comment by probonopd** [comment]
It seems that just like helloSystem (or even more so), Airyx wants to be appealing to switchers coming from the Mac.

As a user coming from the Mac, I find all F... keys decidedly "un-Mac-like", Windows-y. I'd love to see Airyx use Qt's `MacKeyboardScheme`.

So that I can press Command-G for "Find Again", rather than the awkward F3.
And Command-W to close a window, rather than the Windows-y Alt-F4.

Regardless of what keyboard is attached.

**Reply by grahamperrin** [reply]
> Command-G … rather than the awkward F3.

How can one keystroke be more awkward than two?

The muscle memory discussion began at <https://github.com/helloSystem/hello/issues/46#issuecomment-749781939> with @lproven

**Reply by grahamperrin** [reply]
> As a user coming from the Mac, I find all F... keys decidedly "un-Mac-like", 

Amongst the main design goals <https://github.com/mszoek/airyx#readme> ([2021-01-31](https://github.com/mszoek/airyx/commit/d4895aba2d59b17368bf5d285ee03cc2f4a6e719#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R11)): 

>> * compatible with Linux binaries

Users coming from Linux might find Apple keyboards decidedly un-Linux-like. 

I began using Macs in 1993, Apple systems (and hardware) were my primary systems for more than two decades. 

**Function keys are not unusual**. The **Apple extended keyboard**, pictured in 1992: 

![image](https://user-images.githubusercontent.com/192271/135784507-10dc8d11-4007-4c9a-bff9-a4e7a806a630.png)

– F1–F4 are, respectively: **undo, cut, copy, paste** – frequently performed functions.

**Reply by probonopd** [reply]
> Users coming from Linux might find Apple keyboards decidedly un-Linux-like.

And global menu bars. And Objective-C. And much of the other stuff this is all about.

**Reply by grahamperrin** [reply]
> And global menu bars. And Objective-C. And much of the other stuff this is all about.

Entirely off-topic from consistent keyboard shortcuts.

**Reply by probonopd** [reply]
> The Apple extended keyboard

...was totally un-Mac-like, a Sculley product ater Steve had left

**Reply by grahamperrin** [reply]
> un-Mac-like,

I despair at comments such as this.

**Comment by mszoek** [comment]
It's impossible to please everyone with any design :)

**airyxOS will implement the standard Mac keyboard shortcuts found in Apple's Human Interface Guidelines** - so the usual things like Cmd-O (Open), Cmd-Z (Undo), Cmd-W (Close Window/Tab), Cmd-Q (Quit), Cmd-, (Preferences), copy/cut/paste/etc will all work in all applications. Or at least in all "blessed" applications. (see below. note to self: need to document this stuff in the wiki)

Yes, users coming from Linux/FreeBSD or Windows will find this different. 🤷🏻‍♀️ However, there's very little consistency in keyboard shortcuts among *nix applications today. The consistency that does exist tends to be among apps within a DE like KDE/Gnome or within a suite like LibreOffice. So users have to deal with remembering the behavior of each app: is it Ctrl-O or Ctrl-Shift-O or Alt-O to open a file? Does Ctrl-W close a tab/window (common) or do something completely different? Is "Redo" bound to Ctrl-Shift-Z or Ctrl-Y?

Thankfully many *nix apps that implement keyboard shortcuts emulate either Mac or Windows today and the two do overlap in some areas (like cut & paste) aside from the modifier key, so it may not be a huge shift.

I don't mind including extra bindings for Alt-F4, F3 and a few other standard Windows-y shortcuts if that will help smooth the transition. They would be optional and could be turned on in a Preference pane.

Re:
>Amongst the main design goals https://github.com/mszoek/airyx#readme (2021-01-31):
>
>>            compatible with Linux binaries

A note about "blessed" applications and Linux/FreeBSD binaries. AiryxOS will continue to support binaries that run in FreeBSD natively or FreeBSD's Linuxulator. This means mostly command-line tools but does include X11 apps. The intent here is mainly for those applications that are not open source and for which there is no native version - games, for example. (There's often better support for Steam games on Linux than on macOS!) This isn't ideal as they will look and feel quite different, but I feel like it's more important to have the applications even if they are not quite seamless. A "blessed" application is one that conforms to our design goals either by being a Cocoa/Carbon app packaged as a bundle or by being a ported *nix app that follows the same guidelines (e.g. think Gimp on macOS). I intend to curate a repository of such apps that will eventually be integrated with the system.

Bottom line: I consider Linux binaries to be a transitional and then a fallback mechanism as we get towards full macOS compatibility. Not meant to be fully integrated, just to fill gaps.

**Reply by grahamperrin** [reply]
Thanks for clarifying. 

> is it Ctrl-O

In my experience, yes. I never encountered the other two. 

> Does Ctrl-W … do something completely different? 

FreeBSD bug [252503 – editors/featherpad use sane, predictable shortcut Control-W to close tabs](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=252503)
